### PR TITLE
add Recaptcha

### PR DESCRIPTION
**Pull request by apanjain**

### CODE REVIEW

File: backend/pages/views.py [hunk 1]
```diff
@@ -60,33 +65,46 @@ class MentorDocsView (viewsets.ModelViewSet):
...
+                'secret': RECAPTCHA_SECRET_KEY,
+                'response': request.data['g-recaptcha-response'],
+                'remoteip': get_client_ip(request),
+            }
+        )
+        if r.json()['success']:
+            send_email(request)
+            return Response(data={'post':'post'},status=status.HTTP_201_CREATED)
+        return Response(data={'error':'ReCAPTCHA not verified.'}, status=status.HTTP_406_NOT_ACCEPTABLE)
+    return Response(data={'post':'post'},status=status.HTTP_404_NOT_FOUND)
```
Comment (hacksdump) [inline backend/pages/views.py:110]: 405 Method not allowed.

File: frontend/src/routes/Queries/contact.js [hunk 2]
```diff
@@ -18,27 +25,41 @@ class Contact extends Component {
...
+                    errmsg:"<div>Your query has been raised<br/>We'll get back to you soon.</div>",
+                    email:'',
+                    query:'',
+                    name:'',
+                });
+            })
+            .catch((error) => {
+                // console.log(error)
+                this.setState({
+                    errmsg:"<div>There was a problem sending your query<br/>Please try again later.</div>"
```
Comment (hacksdump) [inline frontend/src/routes/Queries/contact.js:56]: This is not being rendered.

File: backend/pages/views.py (hunk 1 above)
Comment (apanjain) [reply backend/pages/views.py]: done

File: frontend/src/routes/Queries/contact.js (hunk 2 above)
Comment (apanjain) [reply frontend/src/routes/Queries/contact.js:56]: flash messages feature will be added in a future pr

**Pull request merged**